### PR TITLE
[SPARK-32877][SQL][TEST] Add test for Hive UDF complex decimal type

### PR DESCRIPTION
**Pull request by ulysses-you**
### What changes were proposed in this pull request?

Add test to cover Hive UDF whose input contains complex decimal type.
Add comment to explain why we can't make `HiveSimpleUDF` extend `ImplicitTypeCasts`.

### Why are the changes needed?

For better test coverage with Hive which we compatible or not.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Add test.

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveUDFSuite.scala [hunk 1]
```diff
@@ -658,6 +658,24 @@ class HiveUDFSuite extends QueryTest with TestHiveSingleton with SQLTestUtils {
     }
   }
 
+  test("SPARK-32877: add test for Hive UDF complex decimal type") {
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveUDFSuite.scala:661]: Thank you for adding this, @ulysses-you . 
Let's trigger with Hive 1.2, too.

**Comment by dongjoon-hyun** [comment]
Retest this please

**Comment by dongjoon-hyun** [comment]
Thank you, @ulysses-you . This PR verified that the newly add test case work in Hive 2.3 and fails with Hive 1.2.
Screen Shot 2020-09-24 at 2 19 25 PM

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveUDFSuite.scala (hunk 1 above)
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveUDFSuite.scala:661]: Please add the following at the first line of the test.
```scala
assume(HiveUtils.isHive23)
```

**Comment by ulysses-you** [comment]
thanks @dongjoon-hyun for the verification ! Added the assume.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @ulysses-you .
Merged to master/3.0.

**Comment by ulysses-you** [comment]
thanks for merging ! FYI @cloud-fan @maropu

**Pull request closed without merging**